src/DataSources/DoctrineDataSource.php

```diff
@@ -0,0 +1,221 @@
+    public function getTotalCount()
+    {
+        return 0;
+    }
+
+
+    /**
+     * Get every single row of the table.
+     * @return array
+     */
+    public function fetchFullData()
+    {
+        $qb = clone $this->qb;
+        return $qb->where([])->setMaxResults(null)->setFirstResult(null)->getQuery()->getResult(Query::HYDRATE_ARRAY);
+        return [];
+    }
+
+
+    /**
+     * Apply limit and offset.
+     * @param  int  $limit   Number of rows
+     * @param  int  $offset  Rows to skip
+     * @return static
+     */
+    public function applyLimit($limit, $offset = 0)
+    {
+        $this->qb->setMaxResults($limit)->setFirstResult($offset);
+        return $this;
+    }
+
+
+    /**
+     * Apply custom WHERE criteria.
+     * @param  string  $column  Column name
+     * @param  array   $custom  Custom criteria
+     * @param  string  $type    Column type
+     * @return static
+     */
+    public function applyCustom($column, array $custom, $type)
+    {
+        return $this;
+    }
+
+
+    /**
+     * @param  string  $column  Column name
+     * @param  array   $custom  Custom criteria
+     * @param  string  $type    Column type
+     * @return static
+     */
+    public function applyCheckers($column, array $value, $type)
+    {
+        return $this;
+    }
+
+
+    /**
+     * Add where condition.
+     * @param  mixed  $args
+     * @return static
+     */
+    public function where($args)
+    {
+        return $this;
+    }
+
+
+    /**
+     * Add ORDER BY directive to the criteria.
+     * @param  string  $column   Column name
+     * @param  string  $sorting  Sorting direction
+     * @return static
+     */
+    public function orderBy($column, $sorting = 'ASC')
+    {
+        $this->qb->autoJoinOrderBy($column, $sorting);
+        return $this;
+    }
+
+
+    /**
+     * Get count with applied where without limit.
+     * @return int
+     */
+    public function count()
+    {
+        return (new Paginator($this->qb))->count();
+    }
+
+
+    /**
+     * Get first element from data.
+     * @return array
+     */
+    public function fetch()
+    {
+        $qb = clone $this->qb;
+        return $qb->setMaxResults(1)->getQuery()->getSingleResult(Query::HYDRATE_ARRAY);
+    }
+
+
+    /**
+     * Get data with applied where, limit and offset.
+     * @return array
+     */
+    public function fetchAll()
+    {
+        return $this->qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
+    }
+
+
+    /**
+     * Get data with applied where without limit and offset.
+     * @return array
+     */
+    public function fetchAllForExport()
+    {
+        return [];
+    }
+
+
+    /**
+     * Get data with applied where, limit and offset and returns tree.
+     * @return array
+     */
+    public function fetchAssoc()
+    {
+		return [];
+    }
+
+
+    /**
+     * Get primary column name.
+     * @return string
+     */
+    public function getPrimaryKey()
+    {
+        return $this->primary_key;
```

Comment:
vytahl bych si z this->qb rootEntity a pak z qb entitymanager, ze ktereho bych si pomoci rootEntity vytahl metadata a pak se jich zeptal na idčko